sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -1923,9 +1926,10 @@ namespace Azure { namespace Storage { namespace Blobs {
         {
           throw std::runtime_error("HTTP status code " + std::to_string(http_status_code));
         }
-        XmlReader reader(
-            reinterpret_cast<const char*>(http_response.GetBodyBuffer().data()),
-            http_response.GetBodyBuffer().size());
+        auto bodyStream = http_response.GetBodyStream();
+        std::vector<uint8_t> bodyContent(bodyStream->Length());
+        bodyStream->Read(&bodyContent[0], bodyContent.size());
+        XmlReader reader(reinterpret_cast<const char*>(bodyContent.data()), bodyContent.size());
```

Comment:
I'm curious why not use `Azure::Core::Http::Reponse::ConstructBodyBufferFromStream` instead for simplicity.

Reply:
I just don't feel like that function.